whole-tale/dashboard
# Added "Connect Additional Accounts" view

## Problem
We need a new view for the "Connect Additional Accounts" functionality.

Fixes #559  
Fixes #560
Fixes #561 

## Approach
* Added a User dropdown to the navbar.
* Added a new Settings view, which is accessible from the User dropdown.

Once the API is live, real data can be wired up. This could either be done as a part of this PR, or as a subsequent one.

### Demonstration
http://recordit.co/YNyn1S1JVj

## How to Test
Prerequisite: must run alongside https://github.com/whole-tale/girder_wholetale/pull/364

1. Checkout and run this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
    * You should see the navbar has a new "user" icon at the top-right
3. Click the "user" icon
    * You should see a dropdown expand
    * The dropdown should contain the user's first/last name and gravatar image
    * The dropdown should provide a link to the Settings page
    * The dropdown should provide an option to Logout
4. Choose the "Settings" nav link
    * You should be brought to the new "Settings" view
    * The view should match the mockups provided, but now uses live data: https://wholetale.org/wholetale-css-mockup/src/settings.html
    * NOTE: DataONE is present, but not yet functional
5. Click "Connect Account" to the right of Zenodo
    * You should see a modal appear allowing you to choose a resource server and enter an API key
6. Choose a resource server, enter an API key, then click Connect
    * The modal should close
    * You should see a new "Authorized on" entry appear beneath Zenodo
    * You should see a red "Disconnect" button beside the "Authorized on" line
7. Click the red Disconnect button
    * You should be greeted with a confirmation modal
8. Confirm the deletion
    * The modal should close
    * You should see the "Authorized on" line disappear
